### ISSUE TITLE
[JENKINS-27948] Script timeout during start of configure a jenkins job

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#551]
<p>I updated jenkins from version 1.596 to version 1.609.<br/>
No I get a script timeout if I try to configure some of the larger jobs.<br/>
I can configure small jobs without any problems.<br/>
I increase the firefox script timeout from 10 seconds to 100 seconds.<br/>
Now I can configure the jenkins jobs, but it takes a very long time to start editing.<br/>
With version 1.596 it was possible to start configuring of the same larger jobs in the predefined timout of 10 seconds</p>

**Attachments:**

- [config.zip](https://issues.jenkins.io/secure/attachment/29457/config.zip)

---
<details><summary><i>Originally reported by <a title="Reiner Wirtz" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-27948" target="_blank">Script timeout during start of configure a jenkins job</a></i></summary>
<i><ul>
<li><b>reported by</b>: rw250155
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-07-11T14:05:54+00:00
<li><b>votes</b>: 6
<li><b>watchers</b>: 14
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Please provide a list of installed plugins and their versions, as well as a sample config.xml of such a 'large job'.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>The list of the installed plugins (copied from display of the system properties):</p>

<p>ant	1.2	true	true<br/>
antisamy-markup-formatter	1.3	true	true<br/>
any-buildstep	0.1	true	false<br/>
artifactdeployer	0.33	true	false<br/>
audit-trail	2.1	true	false<br/>
backup	1.6.1	true	false<br/>
batch-task	1.17	true	false<br/>
build-blocker-plugin	1.6	true	false<br/>
build-environment	1.4	true	false<br/>
build-pipeline-plugin	1.4.7	true	false<br/>
build-user-vars-plugin	1.4	true	false<br/>
clone-workspace-scm	0.6	true	false<br/>
conditional-buildstep	1.3.3	true	false<br/>
config-file-provider	2.7.5	true	false<br/>
copy-to-slave	1.4.4	true	false<br/>
copyartifact	1.35	true	false<br/>
credentials	1.22	true	true<br/>
custom-tools-plugin	0.4.4	true	false<br/>
CustomHistory	1.6	true	false<br/>
cvs	2.12	true	true<br/>
dashboard-view	2.9.4	true	false<br/>
delivery-pipeline-plugin	0.8.11	true	false<br/>
dependencyanalyzer	0.7	true	false<br/>
deploy	1.10	true	false<br/>
disk-usage	0.25	true	false<br/>
downstream-buildview	1.9	true	false<br/>
downstream-ext	1.8	true	false<br/>
dynamic-axis	1.0.3	true	false<br/>
dynamic_extended_choice_parameter	1.0.1	true	false<br/>
dynamicparameter	0.2.0	true	false<br/>
email-ext	2.39.3	true	false<br/>
envfile	1.2	true	false<br/>
envinject	1.91.2	true	false<br/>
extended-choice-parameter	0.35	true	false<br/>
extensible-choice-parameter	1.2.2	true	false<br/>
external-monitor-job	1.4	true	true<br/>
flexible-publish	0.15.1	true	false<br/>
git	2.3.5	true	false<br/>
git-client	1.16.1	true	false<br/>
git-server	1.6	true	false<br/>
groovy	1.24	true	false<br/>
groovy-label-assignment	1.1.0	true	false<br/>
groovyaxis	0.3	true	false<br/>
ivy	1.24	true	false<br/>
javadoc	1.3	true	true<br/>
job-log-logger-plugin	1.0	true	false<br/>
job-node-stalker	1.0.3	true	false<br/>
jobConfigHistory	2.10	true	false<br/>
jquery	1.11.2-0	true	false<br/>
jquery-ui	1.0.2	true	false<br/>
junit	1.5	true	true<br/>
labeled-test-groups-publisher	1.2.7	true	false<br/>
ldap	1.11	true	true<br/>
m2-repo-reaper	1.0	true	false<br/>
mailcommander	1.0.0	true	false<br/>
mailer	1.15	true	true<br/>
managed-scripts	1.1.1	true	false<br/>
mantis	0.26	true	false<br/>
mapdb-api	1.0.6.0	true	false<br/>
matrix-auth	1.2	true	true<br/>
matrix-combinations-parameter	1.0.7	true	false<br/>
matrix-project	1.4.1	true	false<br/>
Matrix-sorter-plugin	1.1	true	false<br/>
maven-plugin	2.9	true	true<br/>
metadata	1.1.0b	true	false<br/>
metrics	3.0.9	true	false<br/>
monitoring	1.55.0	true	false<br/>
nodelabelparameter	1.5.1	true	false<br/>
pam-auth	1.2	true	true<br/>
parameterized-trigger	2.26	true	false<br/>
periodicbackup	1.3	true	false<br/>
postbuild-task	1.8	true	false<br/>
postbuildscript	0.17	true	false<br/>
PrioritySorter	2.11	true	false<br/>
role-strategy	2.2.0	true	false<br/>
run-condition	1.0	true	false<br/>
scm-api	0.2	true	false<br/>
script-security	1.13	true	false<br/>
scriptler	2.7	true	false<br/>
selection-tasks-plugin	1.0	true	false<br/>
shared-objects	0.44	true	false<br/>
show-build-parameters	1.0	true	false<br/>
skip-certificate-check	1.0	true	false<br/>
sonar	2.2	true	false<br/>
ssh-credentials	1.11	true	true<br/>
ssh-slaves	1.9	true	true<br/>
statusmonitor	1.3	true	false<br/>
strawboss	1.3	true	false<br/>
subversion	2.5	true	true<br/>
support-core	2.20	true	false<br/>
svn-tag	1.17	true	false<br/>
testlink	3.10	true	false<br/>
throttle-concurrents	1.8.4	true	false<br/>
token-macro	1.10	true	false<br/>
translation	1.12	true	true<br/>
uno-choice	0.24	true	false<br/>
update-sites-manager	1.0.1	true	false<br/>
validating-string-parameter	2.3	true	false<br/>
vsphere-cloud	2.4	true	false<br/>
weblogic-deployer-plugin	3.2	true	false<br/>
websphere-deployer	1.3-SNAPSHOT (private-08/05/2014 09:34-nothhard)	true	false<br/>
windows-slaves	1.0	true	false<br/>
ws-cleanup	0.25	true	false<br/>
xunit	1.94	true	false</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>The attached zip file contains the config.xml file from a job called DOXiS4_CSB-Test-All-Comb".<br/>
Opening this job for configuration exceeds the limit of 100 seconds configured in my firefox-browser.<br/>
During this time one core of the quad-core-cpu is busy for the whole time.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>I increased the timeout value for scripts to run to 200 seconds.<br/>
The start to configure the job I added in the attachments took 180 seconds (yes 3 minutes) on my machine.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>If I should present the data in a special way or format, please feel free to contact me.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Haven't had time to investigate, but you could do some of that yourself:<br/>
While the form is submitted and the page is loading, repeatedly load the <tt>/threadDump</tt> URL and look for a thread named <tt>Handling POST /job/JOBNAME/configSubmit</tt> or something similar. Look where it's stuck processing for most of the time.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>Excuse my late answer.<br/>
I am glad that I can help.<br/>
I started configuring the job I added in this call as example.<br/>
The browser was about 3 minutes in the state loading data.<br/>
At this time a call to the threadDump URL does not show a thread "Handling POST" or "configSubmit".<br/>
The only Thread with Hanling was the threadDump itself.</p>

<p>It look like jenkins generates Javascript, which runs all the time in the browser.<br/>
It seems it runs a lot longer, if there a more objects or lines to display.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>Skript: <a href="http://sst-bld-master.sst.loca" class="external-link" target="_blank" rel="nofollow noopener">http://sst-bld-master.sst.loca</a>…/0da8a904/scripts/prototype.js:2188</p>

<p>It seems the script is calculating the size of the objects to display.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>Depending on the time, I configure as timeout value for the time a script may run, there are different javascripts active at the time the browser stops the script.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=raspy">raspy</a>:</i>
<p>I observe the same issue also in a different scenario: being in a job with a large list of completed builds, clicking on 'more...' at the bottom of the build list results in the same annoying popups with timeouts in prototype.js. Firefox developer tools console logs:</p>

<p>Error: Script terminated by timeout at:<br/>
Element.Methods.removeClassName@<a href="https://server:8080/static/63938483/scripts/prototype.js:2371:9" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/prototype.js:2371:9</a><br/>
getElementOverflowParams@<a href="https://server:8080/static/63938483/scripts/hudson-behavior.js:1952:5" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/hudson-behavior.js:1952:5</a><br/>
checkRowCellOverflows@<a href="https://server:8080/static/63938483/scripts/hudson-behavior.js:1693:37" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/hudson-behavior.js:1693:37</a><br/>
checkAllRowCellOverflows@<a href="https://server:8080/static/63938483/scripts/hudson-behavior.js:1883:13" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/hudson-behavior.js:1883:13</a><br/>
updateBuilds/<.onSuccess@<a href="https://server:8080/static/63938483/scripts/hudson-behavior.js:1927:21" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/hudson-behavior.js:1927:21</a><br/>
Ajax.Request<.respondToReadyState@<a href="https://server:8080/static/63938483/scripts/prototype.js:1655:10" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/prototype.js:1655:10</a><br/>
Ajax.Request<.onStateChange@<a href="https://server:8080/static/63938483/scripts/prototype.js:1600:7" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/prototype.js:1600:7</a><br/>
bind/<@<a href="https://server:8080/static/63938483/scripts/prototype.js:414:14" class="external-link" target="_blank" rel="nofollow noopener">https://server:8080/static/63938483/scripts/prototype.js:414:14</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<blockquote><p>same issue also in a different scenario</p></blockquote>

<p>In other words, a completely different issue: <a href="https://issues.jenkins.io/browse/JENKINS-26445" title="Browser freezes when clicking more link on a job to see other builds" class="issue-link" data-issue-key="JENKINS-26445"><del>JENKINS-26445</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=raspy">raspy</a>:</i>
<p>Thank you, I could not find it <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> However, I actually get both issues, <a href="https://issues.jenkins.io/browse/JENKINS-26445" title="Browser freezes when clicking more link on a job to see other builds" class="issue-link" data-issue-key="JENKINS-26445"><del>JENKINS-26445</del></a> and <a href="https://issues.jenkins.io/browse/JENKINS-27948" title="Script timeout during start of configure a jenkins job" class="issue-link" data-issue-key="JENKINS-27948"><del>JENKINS-27948</del></a>, so I would suggest they might be related.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=yeongsheng_tan">yeongsheng_tan</a>:</i>
<p>Suggestion above by @scotthebert works.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=chriscralle">chriscralle</a>:</i>
<p>There is a problem with CodeMirror and textarea windows.<br/>
I just upgraded to LTS 1.642.3 this weekend and getting a little rubbing from my comrades about how long the configure load times are.<br/>
I have one 2x4 Matrix multi-step job with conditionals that have shell, python, and windows batch executes.  The configure page for this job generates 8-10 timeouts in codemirror.js and takes minutes to load.<br/>
Common timeouts are on lines 1247, 1207, 853<br/>
These deal with the gutter and scrollbar as far as I can tell.</p>

<p>I've spent Saturday trying everything I can to somehow just DISABLE codemirror. But since it's core, not a plugin, and there's no exposed configuration for it, I'm pretty much stuck with it's behavior.  Is it possible to turn it off?<br/>
The setting in security for syntax highlighting is not related, the number of builds or history is not related.  This is purely codemirror barfing on complex jobs with recursive textareas.</p>

<p>I'm surprised codemirror is not a tag/label on this ticket.</p>

<p>I am creating a sanitized config.xml which I hope to upload to this ticket in the hope that it helps with a resolution.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cmoroexpedia">cmoroexpedia</a>:</i>
<p>I followed the instructions from Scott (Global Security -> In the Markup Formatter section, ensure: Safe HTML is selected and Click on Disable syntax highlighting) and now I can configure jobs in Firefox, but Chrome and Safari are still broken. At least we have a workaround.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hai_lh">hai_lh</a>:</i>
<p>Any idea on how to turn off codemirror to bypass this issue?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rw250155">rw250155</a>:</i>
<p>Hello,</p>

<p>it seems that the bug is fixed.<br/>
I have two jobs, which took about 10 minutes to open for edit with jenkins up to 2.11<br/>
Since I use 2.12 the same jobs takes about 10 to 20 seconds to open for edit.<br/>
And the browser does'n allocated about 10 GB of memory to open the job any more.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally related to: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-32027">JENKINS-32027</a>]</i>
